experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -392,11 +395,23 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		);
 	}
 
+	public getTreeStatus(): TreeStatus {
+		const path = this.cursor.getFieldPath().parent;
+		if (path === undefined) {
+			return TreeStatus.Deleted;
+		}
```

Comment:
I don't think this is the right approach to detecting deleted for a few reasons:
1. if the editable tree node is deleted, accessing its cursor is invalid and ideally would error.
2. This will return true for any root field (the special root or any other detached field)
3. If using the path information, the same data from the anchor instead of cursor will be faster.

For the deleted checking, I recommend adding a method to ProxyTarget, "isFreed" which returns true iff free has been called. TO avoid having to tract ans extra property for that, it can compute it by checking this.lazyCursor's status.

Reply:
Also, given that your implementation aparently has a bug (incorrectly reports deleted for root fields, and doesn't report deleted correctly), I'm guessing it could use some more tests to ensure it is and remains correct in these aspects.

Reply:
Added `isFreed` method to `ProxyTarget` and updated logic for `TreeStatus.Delete`. Added unit tests to check `treeStatus` method